### PR TITLE
bpo-38530: Properly extend UnboundLocalError from NameError

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue38530

### CODE REVIEW

**Review by isidentical — APPROVED** [review]
LGTM

File: Objects/exceptions.c [hunk 1]
```diff
@@ -1393,7 +1393,8 @@ ComplexExtendsException(PyExc_Exception, NameError,
 /*
  *    UnboundLocalError extends NameError
  */
-SimpleExtendsException(PyExc_NameError, UnboundLocalError,
+
+MiddlingExtendsException(PyExc_NameError, UnboundLocalError, NameError,
                        "Local name referenced but not bound to a value.");
```
Comment (pablogsal) [inline Objects/exceptions.c:1398]:
```suggestion
                        "Local name referenced but not bound to a value.");
```

**Review by vstinner — COMMENTED** [review]
issubclass(UnboundLocalError, NameError) already returns True. What is the effect of this change?

**Review by vstinner — APPROVED** [review]
LGTM. Ah, this change is needed because NameError now defines its init, dealloc, traverse, etc. methods. Ok, I get it ;-)

**Pull request merged**